### ISSUE TITLE
syslog-ng config : conditional include 

### ISSUE THREAD

**Issue by lbudai** [balabit/syslog-ng#710]
What do you think about the following extension to the grammar file?

`@include-when` `expression`
`config-snippet`

`expression` could be
* `mod(module)` : true if `module` exists
*  `variable(value)` : true if the `variable` has `value` (eg.: `OS(linux)` )

eg.: (original config is from @faxm0dem )
```
source s_mbox {
  channel {
    source {
      file(
        "/var/spool/mail/fwernli"
        log-msg-size(100000000)
        log-fetch-limit(1)
        flags(no-parse)
        multi-line-mode(regexp)
        multi-line-prefix('^From ')
      );
    };
    @include-when mod(python) {
    parser {
      json-parser(
        template("$(python mbox ${MSG})")
        prefix('email.')
      );
    };
   }; #end include-when
  };
};
```

**Comment by pasztor** [comment]
+1
Very nice solution for the mbox-python scl problem!

**Comment by pasztor** [comment]
Maybe it would simplify the solution, if the 3rd parameter would not be a block to parse, but another filename, like at the @include macro. eg.:
@include-when mod(python) "/usr/local/lib/mbox-parser.inc"

But I do not know the parser that deep. It's just an idea.

**Comment by algernon** [comment]
I agree with @pasztor: it is a bit confusing to have `@include` include a file, but `@include-when` take a block instead. I'd suggest `@when` (and similarly, `@unless`) that take blocks, and `@include-when` (and `@include-unless`) that takes a filename. To be honest, if we have `@when`/`@unless`, then the `@include-$foo` stuff is not needed, either:

`@when mod(python) @include "blah"` would be fine too.

**Comment by faxm0dem** [comment]
:+1: for the file method

**Comment by lbudai** [comment]
Thanks for the feedbacks! Then we will include file and not a block.
@algernon: we will consider your suggestion, thx

**Comment by bazsi** [comment]
Although I see a couple of usecases where conditionals would be useful, but
not in the mbox case. We should make the whole thing either available in
case Python is there or not define the mbox source at all. It should always
behave the same.

Also, what I mentioned IRL as a solution would only work on whole files and
not within blocks. Doing it so that it works within blocks is a lot more
complicated than doing it at the file scope.

File scope is enough to skip the entire mbox include file.

Block scope would be useful to do conditionals based on block arguments,
this is a great feature, but I am afraid it is a lot more difficult.

**Comment by lbudai** [comment]
This issue can be closed, see #827 .
